base_layer/core/src/base_node/sync/header_sync/synchronizer.rs

```diff
@@ -245,59 +249,69 @@ impl<'a, B: BlockchainBackend + 'static> HeaderSynchronizer<'a, B> {
+            .determine_sync_status(&sync_peer.to_string(), best_header, best_block, peer_response.clone())
+            .await?;
+
+        match header_sync_status.clone() {
+            HeaderSyncStatus::InSyncOrAhead => {
+                if best_block_height < best_header_height {
                     debug!(
                         target: LOG_TARGET,
                         "Headers are in sync at height {} but tip is {}. Proceeding to archival/pruned block sync",
-                        header_tip_height,
-                        metadata.height_of_longest_chain()
+                        best_header_height,
+                        best_block_height
                     );
-                    Ok(())
+
+                    Ok(AttemptSyncResult {
+                        headers_returned: peer_response.headers.len() as u64,
+                        fork_hash_index: peer_response.fork_hash_index,
+                        header_sync_status,
+                    })
                 } else {
-                    // Check if the metadata that we had when we decided to enter header sync is behind the peer's
-                    // claimed one. If so, our chain has updated in the meantime and the sync peer
-                    // is behaving.
-                    if self.local_metadata.accumulated_difficulty() <=
-                        sync_peer.claimed_chain_metadata().accumulated_difficulty()
-                    {
-                        debug!(
-                            target: LOG_TARGET,
-                            "Local blockchain received a better block through propagation at height {} (was: {}). \
-                             Proceeding to archival/pruned block sync",
-                            metadata.height_of_longest_chain(),
-                            self.local_metadata.height_of_longest_chain()
-                        );
-                        return Ok(());
-                    }
-                    debug!(
+                    // We will only attempt sync if the our accumulated difficulty is less than the peer's claimed
+                    // accumulated difficulty, thus this is adverse behaviour form the peer.
+                    warn!(
                         target: LOG_TARGET,
                         "Headers and block state are already in-sync (Header Tip: {}, Block tip: {}, Peer's height: \
-                         {})",
-                        header_tip_height,
-                        metadata.height_of_longest_chain(),
+                         {}), peer has lied about chain metadata or did not want to provide headers",
+                        best_header_height,
+                        best_block_height,
                         sync_peer.claimed_chain_metadata().height_of_longest_chain(),
                     );
+
                     Err(BlockHeaderSyncError::PeerSentInaccurateChainMetadata {
                         claimed: sync_peer.claimed_chain_metadata().accumulated_difficulty(),
```

Comment:
This seems like an assumption, the logic here is incorrect. See above comment

Reply:
As per above